girder/api/v1/system.py

```diff
@@ -171,103 +142,108 @@ def getVersion(self, params):
         return version
 
     @access.admin
-    @describeRoute(
+    @autoDescribeRoute(
         Description('Set the list of enabled plugins for the system.')
         .responseClass('Setting')
         .notes('Must be a system administrator to call this.')
-        .param('plugins', 'JSON array of plugins to enable.')
+        .jsonParam('plugins', 'JSON array of plugins to enable.', requireArray=True)
         .errorResponse('Required dependencies do not exist.', 500)
         .errorResponse('You are not a system administrator.', 403)
     )
-    def enablePlugins(self, params):
-        self.requireParams('plugins', params)
-        try:
-            plugins = json.loads(params['plugins'])
-        except ValueError:
-            raise RestException('Plugins parameter should be a JSON list.')
-
+    def enablePlugins(self, plugins, params):
         return self.model('setting').set(SettingKey.PLUGINS_ENABLED, plugins)
 
     @access.admin
-    @describeRoute(
+    @autoDescribeRoute(
         Description('Unset the value for a system setting.')
-        .notes("""Must be a system administrator to call this. This is used to
-               explicitly restore a setting to its default value.""")
+        .notes('Must be a system administrator to call this. This is used to '
+               'explicitly restore a setting to its default value.')
         .param('key', 'The key identifying the setting to unset.')
         .errorResponse('You are not a system administrator.', 403)
     )
-    def unsetSetting(self, params):
-        self.requireParams('key', params)
-        return self.model('setting').unset(params['key'])
+    def unsetSetting(self, key, params):
+        return self.model('setting').unset(key)
 
     @access.admin(scope=TokenScope.PARTIAL_UPLOAD_READ)
-    @describeRoute(
+    @autoDescribeRoute(
         Description('Get a list of uploads that have not been finished.')
         .notes('Must be a system administrator to call this.')
         .param('uploadId', 'List only a specific upload.', required=False)
         .param('userId', 'Restrict listing uploads to those started by a '
                'specific user.', required=False)
         .param('parentId', 'Restrict listing uploads to those within a '
                'specific folder or item.', required=False)
-        .param('assetstoreId', 'Restrict listing uploads within a specific '
-               'assetstore.', required=False)
+        .param('assetstoreId', 'Restrict listing uploads within a specific assetstore.',
+               required=False)
         .param('minimumAge', 'Restrict listing uploads to those that are at '
-               'least this many days old.', required=False)
+               'least this many days old.', dataType='float', required=False)
```

Comment:
Better as an integer (yes, from the caller's perspective it is just clearer what you are getting)?  Would that break backwards compatibility?